### ISSUE TITLE
assigning partial charges fails if conformer generation fails

### ISSUE THREAD

**Issue by richardjgowers** [openforcefield/openff-interchange#829]
**Description**

I've got reports of multiple simulations failing when conformer generation fails as part of the partial charge assignment.

**Reproduction**

I think the actual error happens in SmirnoffTemplateGenerator, but I think this is now running through Interchange(?)

```python
ff = ForceField('openff-2.0.0.offxml')
cubic_box = unit.Quantity(30 * np.eye(3), unit.angstrom)

rdk_mol = Chem.MolFromSmiles('CC')
off_mol = toolkit.topology.Molecule(rdk_mol)

with mock.patch('rdkit.Chem.AllChem.EmbedMultipleConfs', return_value=0) as m:
    i = interchange.Interchange.from_smirnoff(topology=[off_mol], force_field=ff, box=cubic_box)
```

**Output**

Please include the output, including full tracebacks of any errors, resulting from the reproduction.

```
ValueError: No registered toolkits can provide the capability "assign_partial_charges" for args "()" and kwargs "{'molecule': Molecule with name xx and SMILES xx

'partial_charge_method': 'am1bcc', 'use_conformers': None, 'strict_n_conformers': False, 'normalize_partial_charges': True, '_cls': <class 'openff.toolkit.topology.molecule.Molecule'>}"

Available toolkits are: [ToolkitWrapper around The RDKit version 2023.03.2, ToolkitWrapper around AmberTools version 22.0, ToolkitWrapper around Built-in Toolkit version None]

ToolkitWrapper around The RDKit version 2023.03.2 <class 'openff.toolkit.utils.exceptions.ChargeMethodUnavailableError'> : partial_charge_method 'am1bcc' is not available from RDKitToolkitWrapper. Available charge methods are ['gasteiger', 'mmff94']

 ToolkitWrapper around AmberTools version 22.0 <class 'openff.toolkit.utils.exceptions.ConformerGenerationError'> : RDKit conformer generation failed.

ToolkitWrapper around Built-in Toolkit version None <class 'openff.toolkit.utils.exceptions.ChargeMethodUnavailableError'> : Partial charge method "am1bcc"" is not supported by the Built-in toolkit. Available charge methods are ['zeros', 'formal_charge']
```

The molecule(s) already has a conformer, so failing on partial charge generator because a conformer couldn't be generated is a little frustrating.  Would it be possible to add some fallback option to use the molecule's existing conformer to assign partial charges?

**Comment by mattwthompson** [comment]
The toolkit lies at the center of the Russian doll; the layers radiating outward are Interchange, `openmmforcefields`, and then whatever calls `SmirnoffTemplateGenerator`. The most straightforward reproduction would be something like this

```python
from unittest import mock

from openff.toolkit import Molecule


molecule = Molecule.from_smiles("CC")

with mock.patch('rdkit.Chem.AllChem.EmbedMultipleConfs', return_value=0) as m:
    molecule.assign_partial_charges(partial_charge_method="am1bcc")
```

As I understand it, the toolkit throwing away conformers is intentional. This happens [here](https://github.com/openforcefield/openff-toolkit/blob/0.14.4/openff/toolkit/utils/ambertools_wrapper.py#L176-L197), just before calling `sqm`. There is the backdoor of `use_conformers` if interacting with the toolkit API directly; this works locally:

```python
molecule = Molecule.from_smiles("CC")
molecule.generate_conformers()

with mock.patch('rdkit.Chem.AllChem.EmbedMultipleConfs', return_value=0) as m:
    try:
        molecule.assign_partial_charges(partial_charge_method="am1bcc")
    except ValueError:
        molecule.assign_partial_charges(partial_charge_method="am1bcc", use_conformers=molecule.conformers)

assert molecule.partial_charges is not None
```

I agree it's a surprising error. I'd be more than a little frustrating to see a conformer generation failure crash a pipeline when my molecule(s) already have conformers. Until just now, I haven't pieced together that this backdoor could be made into default behavior. I don't read the [SMIRNOFF specification](https://openforcefield.github.io/standards/standards/smirnoff/#toolkitam1bcc-temporary-support-for-toolkit-based-am1-bcc-partial-charges) to have any guidance on what should happen here:

> This tag calculates partial charges using the default settings of the highest-priority cheminformatics toolkit that can perform [AM1-BCC charge assignment](https://docs.eyesopen.com/toolkits/python/quacpactk/molchargetheory.html#am1bcc-charges). Currently, if the OpenEye toolkit is licensed and available, this will use QuacPac configured to generate charges using [AM1-BCC ELF10](https://docs.eyesopen.com/toolkits/python/quacpactk/OEProtonClasses/OEAM1BCCELF10Charges.html) for each unique molecule in the topology. Otherwise [RDKit](https://rdkit.org/) will be used for initial conformer generation and the [AmberTools antechamber/sqm software](https://ambermd.org/AmberTools.php) will be used for charge calculation.

> If this tag is specified for a force field, conformer generation will be performed regardless of whether conformations of the input molecule were provided. If RDKit/AmberTools are used as the toolkit backend for this calculation, only the first conformer is used for AM1-BCC calculation.

It's easy to see a strong case for falling back to user-generated conformers by default. This would be a major behavior change ... for what I believe to be a tiny subset of inputs. The implementation should be straightforward[^1] I'll need to think about this a little more before committing to it ... I wonder if this is something that SMIRNOFF should have something to say about. In the mean time, I'll give @j-wags the opportunity to provide a hard veto.

[^1]: Provided a molecule that reliably fails conformer generation but doesn't have other insane things going on - IME this has been hard for me to come by.

**Comment by j-wags** [comment]
Yeah, I agree with Matt's response. The root of this is in the Toolkit so I'll move it over to that tracker. 

The "making a new conformer for charge assignment" behavior is a deliberate feature. It keeps us closer to our goal of assigning parameters based only on the molecular graph. This philosophy isn't perfectly implemented - OE and RDKit+AmberTools are different implementations and so their conformer generation and charge assignment can vary - but it should at least give the same charges if a user uses the same molecular graph twice in one day. Which wouldn't be the case if we used the input geometry.

I'd prefer not to silently fall back to user conformers here, and I think that making the users specify `use_conformers` for the charge assignment is an appropriate acknowledgement that the force field is being applied in a nonstandard way.

I'm surprised that there's a molecule suitable for AM1BCC that RDKit can't make a conformer for, and I wonder if this is really a sanitization issue in disguise. Could you share the input that causes this error?
